cmd/ci-secret-bootstrap/main.go

```diff
@@ -0,0 +1,297 @@
+	fs.StringVar(&o.bwPasswordPath, "bw-password-path", "", "Path to a password file to access BitWarden.")
+	if err := fs.Parse(os.Args[1:]); err != nil {
+		logrus.WithError(err).Errorf("cannot parse args: %q", os.Args[1:])
+	}
+	return o
+}
+
+func (o *options) validateOptions() error {
+	if o.bwUser == "" {
+		return fmt.Errorf("--bw-user is empty")
+	}
+	if o.bwPasswordPath == "" {
+		return fmt.Errorf("--bw-password-path is empty")
+	}
+	if o.configPath == "" {
+		return fmt.Errorf("--config is empty")
+	}
+	return nil
+}
+
+func (o *options) completeOptions() error {
+	bytes, err := ioutil.ReadFile(o.bwPasswordPath)
+	if err != nil {
+		return err
+	}
+	o.bwPassword = strings.TrimSpace(string(bytes))
+
+	kubeConfigs, _, err := util.LoadKubeConfigs(o.kubeConfigPath)
+	if err != nil {
+		return err
+	}
+
+	bytes, err = ioutil.ReadFile(o.configPath)
+	if err != nil {
+		return err
+	}
+	var config []secretConfig
+	err = yaml.Unmarshal(bytes, &config)
+	if err != nil {
+		return err
+	}
+	o.config = config
+
+	return o.validateCompletedOptions(kubeConfigs)
+}
+
+func (o *options) validateCompletedOptions(kubeConfigs map[string]rest.Config) error {
+	o.secretsGetters = map[string]coreclientset.SecretsGetter{}
+	if o.bwPassword == "" {
+		return fmt.Errorf("--bw-password-file was empty")
+	}
+	if len(o.config) == 0 {
+		return fmt.Errorf("--config was empty")
+	}
+	toMap := map[string]map[string]string{}
+	for i, secretConfig := range o.config {
+		if len(secretConfig.From) == 0 {
+			return fmt.Errorf("config[%d].from is empty", i)
+		}
+		if len(secretConfig.To) == 0 {
+			return fmt.Errorf("config[%d].to is empty", i)
+		}
+		for key, bwContext := range secretConfig.From {
+			if key == "" {
+				return fmt.Errorf("config[%d].from: empty key is not allowed", i)
+			}
+			if bwContext.BWItem == "" {
+				return fmt.Errorf("config[%d].from[%s]: empty value is not allowed", i, key)
+			}
+			if bwContext.Field == "" && bwContext.Attachment == "" {
+				return fmt.Errorf("config[%d].from[%s]: either field or attachment needs to be non-empty", i, key)
+			}
+			if bwContext.Field != "" && bwContext.Attachment != "" {
+				return fmt.Errorf("config[%d].from[%s]: cannot use both field and attachment", i, key)
+			}
+		}
+		for j, secretContext := range secretConfig.To {
+			if secretContext.Cluster == "" {
+				return fmt.Errorf("config[%d].to[%d].cluster: empty value is not allowed", i, j)
+			}
+			if secretContext.Namespace == "" {
+				return fmt.Errorf("config[%d].to[%d].namespace: empty value is not allowed", i, j)
+			}
+			if secretContext.Name == "" {
+				return fmt.Errorf("config[%d].to[%d].name: empty value is not allowed", i, j)
+			}
+			if _, ok := kubeConfigs[secretContext.Cluster]; !ok {
+				return fmt.Errorf("config[%d].to[%d]: failed to find cluster context %q in the kubeconfig", i, j, secretContext.Cluster)
+			}
+
+			kc := kubeConfigs[secretContext.Cluster]
```

Comment:
This is still here and still not validation. Validation should do validation. Create your clients elsewhere.

Reply:
Moved to `completeOptions`.